src/libs/actions/Report.js

```diff
@@ -797,8 +797,17 @@ function subscribeToReportCommentPushNotifications() {
 
     // Open correct report when push notification is clicked
     PushNotification.onSelected(PushNotification.TYPE.REPORT_COMMENT, ({reportID}) => {
-        Navigation.setDidTapNotification();
-        Linking.openURL(`${CONST.DEEPLINK_BASE_URL}${ROUTES.getReportRoute(reportID)}`);
+        if (Navigation.isReady()) {
+            // If a chat is visible other than the one we are trying to navigate to, then we need to navigate back
+            if (Navigation.isChatVisible() && Navigation.isActiveRoute(`r/${reportID}`)) {
```

Comment:
`Navigation.navigate` wouldn't go to another report while on a report, but this fixes it. Open to discussion on a more elegant solution using the existing functions if one is available.